helsingborg-stad/app-mitt-helsingborg
# Feat/summary list validation

## Explain the changes you’ve made

Add logic to handle validation to the summary list

## Explain why these changes are made

So that the input fields inside the summary handles validation in the same way as other input fields. Right now they only display a red border when something is wrong.

## Explain your solution

First, I added some logic to the formReducer for storing a list of all questions. Previously, the validation acted on just the questions of the current step, but for the summary field we need to look at all questions since the value might not belong to a question on the same step. I.e., the summary list refers to questions elsewhere in the form, and to get the validation rules, I need to have access to the question. So I create a function in formAction to do this, and dispatch an action when the form is loaded to trigger it.

The way the validation errors are sent from Step to the FormFields are refactored slightly, again since the summary list needs access to all validation errors of the form. The object with all errors are sent down to the summary list, which in turn sends the right error through to the SummaryListItem component. This component in turn renders the red border or not, depending. 

Also added a story, Summary list validation, that loads a form and shows how the validation works. The PR is a bit big, but this story is like 280 lines, so it actually accounts for most of it.

(I removed the 'countMainSteps function from useForm, since it was duplicated in formActions. Not related to the functionality, so sorry)

## How to test the changes?

Go into storybook and into the Summary list validation story, fill out some values and check that the summary list inputs are correctly validated. 
Or load a form in the app which has a summary list, like the test form in develop. 

## Was this feature tested in the following environments?
- [] Building the Application on a Android device/simulator.
